injection/sharedmain/main.go

```diff
@@ -111,6 +113,38 @@ var (
 	WebhookMainWithConfig  = MainWithConfig
 )
 
+// MainNamed runs the generic main flow for controllers and webhooks.
+func MainNamed(ctx context.Context, component string, ctors ...injection.NamedControllerConstructor) {
```

Comment:
The HA flag will be removed soon since HA has been stable for a really long time.

I'm not sure how what you're suggesting should work.

What `WithNamedControllerConstructor(ctors ...injection.NamedControllerConstructor) Option` should be doing?

Reply:
I don't really like pushing and pulling things from the context in general, especially if we can have proper typing in the interface.